stores/sql/mysql/main.go

```diff
@@ -566,6 +570,63 @@ func (tx *MainDatabaseTx) ProcessChainUpdate(ctx context.Context, fn func(ssql.C
 	})
 }
 
+func (tx *MainDatabaseTx) PrunableContractRoots(ctx context.Context, fcid types.FileContractID, roots []types.Hash256) (indices []uint64, err error) {
+	// build tmp table name
+	var r [4]byte
+	frand.Read(r[:])
+	tmpTable := strings.ReplaceAll(fmt.Sprintf("diff_%s_%x", fcid.String()[:8], r), ":", "_")
+
+	// create temporary table
+	_, err = tx.Exec(ctx, fmt.Sprintf(`
+CREATE TEMPORARY TABLE %s (idx INT, root blob);
+CREATE INDEX %s_idx ON %s (root(32));`, tmpTable, tmpTable, tmpTable))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temporary table: %w", err)
+	}
+
+	// defer removal
+	defer func() {
+		tx.Exec(ctx, fmt.Sprintf(`
+DROP TABLE IF EXISTS %s;
+DROP INDEX IF EXISTS %s_idx;`, tmpTable, tmpTable))
```

Comment:
I think the index is dropped implicitly. I'd also add a `DROP TABLE IF EXISTS` before the query in case dropping the table afterwards fails for whatever reason. 
For the drop afterwards I'd not add the `IF EXISTS` since we know it should exist and log the error

Reply:
Ok, the tmp table has a random part though, are you proposing to remove the random part?

Reply:
Yeah we shouldn't need it. Maybe a unique name per contract is good but random is probably a bit too much.